static/app/views/settings/organizationAuthTokens/index.spec.tsx

```diff
@@ -0,0 +1,288 @@
+
+    projectsMock = MockApiClient.addMockResponse({
+      url: PROJECTS_ENDPOINT,
+      method: 'GET',
+      body: [project],
+    });
+  });
+
+  afterEach(function () {
+    MockApiClient.clearMockResponses();
+  });
+
+  it('shows tokens', async function () {
+    const tokens: OrgAuthToken[] = [
+      {
+        id: '1',
+        name: 'My Token 1',
+        tokenLastCharacters: '1234',
+        dateCreated: new Date('2023-01-01T00:00:00.000Z'),
+        scopes: ['org:read'],
+      },
+      {
+        id: '2',
+        name: 'My Token 2',
+        tokenLastCharacters: 'ABCD',
+        dateCreated: new Date('2023-01-01T00:00:00.000Z'),
+        scopes: ['org:read'],
+        dateLastUsed: new Date(),
+        projectLastUsedId: project.id,
+      },
+    ];
+
+    const mock = MockApiClient.addMockResponse({
+      url: ENDPOINT,
+      method: 'GET',
+      body: tokens,
+    });
+
+    render(<OrganizationAuthTokensIndex {...defaultProps} />);
+
+    // First shows loading indicator
+    await screen.findByTestId('loading-indicator');
+
+    // Then list
+    await screen.findByText('My Token 1');
+    screen.getByText('My Token 2');
+    screen.getByText('never used');
+    await screen.findByText(
+      textWithMarkupMatcher('a few seconds ago in project Project Name')
+    );
+    expect(screen.queryByTestId('loading-error')).not.toBeInTheDocument();
+    expect(screen.queryByTestId('loading-indicator')).not.toBeInTheDocument();
+    expect(screen.queryByTestId('empty-state')).not.toBeInTheDocument();
+
+    await waitFor(() => {
+      expect(mock).toHaveBeenCalledTimes(1);
+      expect(mock).toHaveBeenCalledWith(
+        ENDPOINT,
+        expect.objectContaining({method: 'GET'})
+      );
+      expect(projectsMock).toHaveBeenCalledTimes(1);
+      expect(projectsMock).toHaveBeenCalledWith(
+        PROJECTS_ENDPOINT,
+        expect.objectContaining({method: 'GET', query: {query: `id:${project.id}`}})
+      );
+    });
+  });
+
+  it('handle error when loading tokens', async function () {
+    const mock = MockApiClient.addMockResponse({
+      url: ENDPOINT,
+      method: 'GET',
+      statusCode: 400,
+    });
+
+    render(<OrganizationAuthTokensIndex {...defaultProps} />);
+
+    await waitFor(() => {
+      expect(mock).toHaveBeenCalledTimes(1);
+    });
+
+    expect(await screen.findByTestId('loading-error')).toHaveTextContent(
+      'Failed to load auth tokens for the organization.'
+    );
+    expect(screen.queryByTestId('loading-indicator')).not.toBeInTheDocument();
+    expect(screen.queryByTestId('empty-state')).not.toBeInTheDocument();
+  });
+
+  it('shows empty state', async function () {
+    const tokens: OrgAuthToken[] = [];
+
+    MockApiClient.addMockResponse({
+      url: ENDPOINT,
+      method: 'GET',
+      body: tokens,
+    });
+
+    render(<OrganizationAuthTokensIndex {...defaultProps} />);
+
+    // First shows loading indicator
+    await screen.findByTestId('loading-indicator');
```

Comment:
do we want to check for this? usually when we have to wait for something else to show up, it is due to a loading being performed

Reply:
My idea for this test was to test the flow:

1. loading indicator is shown
2. loading is completed, data is shown - loading indicator is hidden

But happy to delete it if we don't think that's a helpful test :)